src/test/java/seedu/address/model/reminder/ReminderOnOffTest.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.model.reminder;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+class ReminderOnOffTest {
+
+    @Test
+    void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new ReminderOnOff(null));
+    }
+
+    @Test
+    void constructor_emptySetReminder_throwsIllegalArgumentException() {
+        String emptySetReminder = "";
+        assertThrows(IllegalArgumentException.class, () -> new ReminderOnOff(emptySetReminder));
+    }
+
+    @Test
+    void constructor_invalidSetReminder_throwsIllegalArgumentException() {
+        String invalidSetReminder = "truee";
+        assertThrows(IllegalArgumentException.class, () -> new ReminderOnOff(invalidSetReminder));
+    }
+
+    @Test
+    void constructor_defaultReminder() {
+        // default reminder is false
+        ReminderOnOff defaultReminder = new ReminderOnOff();
+        assertFalse(defaultReminder.getReminder());
+    }
+
+    @Test
+    void isValidSetReminder() {
+        // invalid set reminder
+        assertFalse(ReminderOnOff.isValidSetReminder("")); // empty string
+        assertFalse(ReminderOnOff.isValidSetReminder(" ")); // spaces only
+        assertFalse(ReminderOnOff.isValidSetReminder("truee")); // incorrect spelling
+        assertFalse(ReminderOnOff.isValidSetReminder("falsee")); // incorrect spelling
+        assertFalse(ReminderOnOff.isValidSetReminder("phone")); // non-boolean
+        assertFalse(ReminderOnOff.isValidSetReminder("1")); // non-boolean
+
+        // valid set reminder
+        assertTrue(ReminderOnOff.isValidSetReminder("true")); // true
+        assertTrue(ReminderOnOff.isValidSetReminder("false")); // false
```

Comment:
You can also consider adding different capitalizations to show that it is case insensitive